components/TextFields/src/ContainedInputView/MDCBaseTextField.m

```diff
@@ -265,15 +315,76 @@ - (CGRect)rightViewRectForBounds:(CGRect)bounds {
 }
 
 - (CGRect)textRectForBounds:(CGRect)bounds {
-  return [self adjustTextAreaFrame:self.layout.textRect
+  CGRect textRect = [self textRectFromLayout:self.layout labelState:self.labelState];
+  return [self adjustTextAreaFrame:textRect
       withParentClassTextAreaFrame:[super textRectForBounds:bounds]];
 }
 
 - (CGRect)editingRectForBounds:(CGRect)bounds {
-  return [self adjustTextAreaFrame:self.layout.textRect
+  CGRect textRect = [self textRectFromLayout:self.layout labelState:self.labelState];
+  return [self adjustTextAreaFrame:textRect
       withParentClassTextAreaFrame:[super textRectForBounds:bounds]];
 }
 
+- (UIFont *)normalFont {
+  return self.font ?: [self uiTextFieldDefaultFont];
```

Comment:
Why not:

return self.font ?: [UIFont systemFontOfSize:[UIFont systemFontSize]];

Reply:
I created `normalFont` because I think (last time I checked) `font` is nullable, and I put the default font in a dispatch_once block because I didn't want to have to initialize it over and over again.